agent/src/main/java/kanela/agent/api/instrumentation/listener/InstrumentationRegistryListener.java

```diff
@@ -16,79 +16,127 @@
 
 package kanela.agent.api.instrumentation.listener;
 
-import io.vavr.Tuple;
-import io.vavr.collection.HashMap;
-import io.vavr.collection.List;
-import io.vavr.collection.Map;
+import static kanela.agent.util.classloader.ScalaCompilerClassLoaderFilter.isScalaCompilerClassLoader;
 import kanela.agent.api.instrumentation.TypeTransformation;
-import kanela.agent.bootstrap.dispatcher.Dispatcher;
 import kanela.agent.util.classloader.ClassLoaderNameMatcher;
 import kanela.agent.util.conf.KanelaConfiguration;
+import lombok.val;
 import net.bytebuddy.agent.builder.AgentBuilder;
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.dynamic.DynamicType;
 import net.bytebuddy.utility.JavaModule;
 
-public class InstrumentationRegistryListener extends AgentBuilder.Listener.Adapter {
 
-    private final static String DISPATCHER_KEY = "InstrumentationRegistryListener";
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
 
-    private Map<String, Map<TypeTransformation, List<TypeDescription>>> moduleTransformers = HashMap.empty();
-    private Map<String, KanelaConfiguration.ModuleConfiguration> modulesConfiguration = HashMap.empty();
-    private Map<String, List<Throwable>> errors = HashMap.empty();
+/**
+ * Captures and exposes the instrumentation modules known by Kanela. Since users of this class might be trying to access
+ * the exposed data from a ClassLoader that is not connected to Kanela's (e.g. when running an application from SBT), we
+ * will only expose data using plain Java types which are guaranteed to be loaded by the Bootstrap ClassLoader.
+ *
+ */
+public class InstrumentationRegistryListener extends AgentBuilder.Listener.Adapter {
```

Comment:
try using @Value that does not bite;) :https://github.com/kamon-io/kanela/blob/master/agent/src/main/java/kanela/agent/api/instrumentation/listener/DefaultInstrumentationListener.java#L28-L29

Reply:
Similar reasons to with the `Kanela` class, it's not a good candidate because it's state is actually mutable. :man_shrugging: